mitmproxy/proxy/layers/http/__init__.py

```diff
@@ -308,6 +309,25 @@ def send_response(self, already_streamed: bool = False):
         """We have either consumed the entire response from the server or the response was set by an addon."""
         assert self.flow.response
         self.flow.response.timestamp_end = time.time()
+
+        is_websocket = (
+            self.flow.response.status_code == 101
+            and
+            self.flow.response.headers.get("upgrade", "").lower() == "websocket"
+            and
+            self.flow.request.headers.get("Sec-WebSocket-Version", "") == "13"
```

Comment:
Should we use wsproto constants here instead?

Reply:
https://github.com/python-hyper/wsproto/blob/55bf7a4ee1e85e3ebc93f04708327b0d0f193925/src/wsproto/handshake.py#L36

Reply:
Yes that's good. We leave it hardcoded in the tests, so we have an obvious breakage if one of them ever changes.